src/pages/login/PasswordModal.tsx

```diff
@@ -14,6 +16,7 @@ interface Props {
 }
 
 const PasswordModal: FC<Props> = ({ onConfirm, onClose }) => {
+  const [passwordRequired, setPasswordRequired] = useState(false);
```

Comment:
The default `false` currently should better be driven by the caller. If the user opens this from the macOS section, the default should be `true`, it can stay `false` in the other cases.